test/Elastic.Apm.Tests/BackendCommTests/PayloadSenderTests.cs

```diff
@@ -329,21 +325,23 @@ internal async void FlushInterval_test(TestArgs args, int numberOfEventsToSend)
 
 			var handler = new MockHttpMessageHandler((r, c) =>
 			{
+				TestOutputHelper.WriteLine("Receiving request in MockHttpMessageHandler");
 				batchSentBarrier.SignalAndWait(barrierTimeout).Should().BeTrue();
 				return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK));
 			});
 
 			var configurationReader = args.BuildConfig(_logger);
 			var service = Service.GetDefaultService(configurationReader, _logger);
-			var payloadSender = new PayloadSenderV2(_logger, configurationReader, service, new Api.System(), MockApmServerInfo.Version710,
-				handler, /* dbgName: */ TestDisplayName);
-
-			using (var agent = new ApmAgent(new TestAgentComponents(_logger, payloadSender: payloadSender)))
+			using (var payloadSender = new PayloadSenderV2(_logger, configurationReader, service, new Api.System(), MockApmServerInfo.Version710,
```

Comment:
I don't really get how this can happen, since `ApmAgent` disposes `TestAgentComponents` which should dispose `PayloadSenderV2`. Am I missing something? 🤔

Reply:
tldr: indeed, this isn't needed, I reverted all changes in this file.

Yeah, I was confused as well, and later on I had another commit in `PayloadSenderV2` which fixed the real issue. Nevertheless, I had test failures in a previous stage where logs showed that `PayloadSender` started being disposed too early and adding the `using` changed that behavior. 

Logs from that stage:

Reply:
Thanks for the clarification.